packages/starknet_provider/lib/src/model/invoke_transaction.dart

```diff
@@ -62,6 +70,29 @@ class InvokeTransactionV1
       _$InvokeTransactionV1FromJson(json);
 }
 
+@freezed
+class InvokeTransactionV3
+    with _$InvokeTransactionV3
+    implements InvokeTransaction {
+  const factory InvokeTransactionV3({
+    @Default('INVOKE') String type,
+    required List<Felt> accountDeploymentData,
+    required List<Felt> calldata,
+    required String feeDataAvailabilityMode,
+    required Felt nonce,
+    required String nonceDataAvailabilityMode,
+    required List<Felt> paymasterData,
+    required Map<String, ResourceBounds> resourceBounds,
```

Comment:
The import of 'declare_transaction.dart' is unnecessary because all of the used elements are also provided by the import of 'package:starknet_provider/starknet_provider.dart'.